deptofdefense/move.mil
# Improve logging

## Checklist

I have…

- ~~~attached screenshots illustrating relevant behavior before and after my changes.~~~

## Summary of Changes

This pull request addresses a concern that location data (latitude and longitude) were being captured in the application's logs (based on Rails' default logging choices). The [Lograge gem](https://github.com/roidrage/lograge) addresses this concern by dramatically paring down the information logged.

### Before adding Lograge

In `log/development.log`:

```
Started GET "/resources/locator-maps?coordinates=38.850107799999996,-77.03917539999999" for 127.0.0.1 at 2017-10-12 09:24:36 -0400
   (1.5ms)  SELECT "schema_migrations"."version" FROM "schema_migrations" ORDER BY "schema_migrations"."version" ASC
Processing by OfficesController#index as HTML
  Parameters: {"coordinates"=>"38.850107799999996,-77.03917539999999"}
  Rendering offices/index.html.erb within layouts/application
  Rendered shared/_sidenav.html.erb (10.6ms)
```

…and in `log/production.log`:

```
I, [2017-10-12T13:39:12.384339 #332]  INFO -- : [61456cf2-933f-410f-8421-743eeb9d48b6] Started GET "/resources/locator-maps?coordinates=38.850107799999996,-77.03917539999999" for 152.179.8.202 at 2017-10-12 13:39:12 +0000
I, [2017-10-12T13:39:12.385213 #332]  INFO -- : [61456cf2-933f-410f-8421-743eeb9d48b6] Processing by OfficesController#index as HTML
I, [2017-10-12T13:39:12.385314 #332]  INFO -- : [61456cf2-933f-410f-8421-743eeb9d48b6]   Parameters: {"coordinates"=>"38.850107799999996,-77.03917539999999"}
I, [2017-10-12T13:39:12.386591 #332]  INFO -- : [61456cf2-933f-410f-8421-743eeb9d48b6]   Rendering offices/index.html.erb within layouts/application
I, [2017-10-12T13:39:12.388754 #332]  INFO -- : [61456cf2-933f-410f-8421-743eeb9d48b6]   Rendered shared/_sidenav.html.erb (1.9ms)
```

### After adding Lograge

In `log/development.log`:

```
method=GET path=/resources/locator-maps format=html controller=OfficesController action=index status=200 duration=2001.12 view=1962.32 db=14.32
```

## Testing

To verify the changes proposed in this pull request…

1. clone this repo,
1. `git checkout improve-logging`,
1. set up development dependencies according to `CONTRIBUTING.md`,
1. run `bin/rails server`,
1. navigate to a page and note that the output in `log/development.log` and in your Terminal is _much_ shorter and does not include parameters.
